src/main/java/com/synopsys/integration/phonehome/request/PhoneHomeRequestBodyBuilder.java

```diff
@@ -33,18 +34,21 @@
     private final String customerId;
     private final String hostName;
     private final NameVersion artifactInfo;
-    private final NameVersion productInfo;
+    private final UniquePhoneHomeProduct product;
+    private final String productVersion;
     private final List<String> artifactModules = new ArrayList<>();
     private final Map<String, String> metaData = new HashMap<>();
 
-    public PhoneHomeRequestBodyBuilder(String customerId, String hostName, NameVersion artifactInfo, NameVersion productInfo) {
+    public PhoneHomeRequestBodyBuilder(String customerId, String hostName, NameVersion artifactInfo, UniquePhoneHomeProduct product, String productVersion) {
+        if (null == product || null == artifactInfo || StringUtils.isAnyBlank(customerId, hostName, artifactInfo.getName(), artifactInfo.getVersion(), product.getName(), productVersion)) {
+            throw new IllegalArgumentException("The fields: customerId, hostName, artifactInfo, and productInfo are all required.");

```

Comment:
Since the NameVersion productInfo no longer exists, this error message should likely either be updated or ProductInfo should be reintroduced as a special kind of NameVersion